### DISCUSSION TITLE
Notes discovery: Random notes and recent notes (dashboard?)

### DISCUSSION THREAD

**Discussion by j-steinbach** [srid/emanote#139] [Ideas]
There should be ways to discover random/recent/orphaned notes. 

- Google offers a "I am feeling lucky" button that takes one to a random page/function. Wikipedia offers a "random article" page
- [Wikipedia](https://en.wikipedia.org/wiki/Main_Page) displays *curiosities* and timeline events on its start page
- Modern browsers offer a custom "new tab" experience, with recent searches

The most important thing here is giving the user choice -- those features shouldn't feel gimmicky, but instead be optional and enhance the side menu or "main entry page" of the personal knowledge base / Zettelkasten

**Comment by j-steinbach** [comment]
Just noticed that there is/was https://github.com/srid/neuron/issues/427

**Reply by srid** [reply]
Have you tried https://github.com/srid/emanote/issues/10 ?

Tag queries already work. You can also query by path, eg: https://inspirativity.zettel.page/

**Reply by j-steinbach** [reply]
So one could create a "random" query and add it to their index note?

**Reply by srid** [reply]
Yes, the Query type can be extended to accommodate such 'dynamic' cases.

https://github.com/srid/emanote/blob/e0229f1b9b3b2fc49a0aab51f098a4211734bea6/src/Emanote/Model/Query.hs#L22-L27

But we need to come up with exact semantics. Is it just "give me a link to a single random note"?

**Reply by j-steinbach** [reply]
I would go with "any random note" and "random note with tag XYZ", for example "random note tagged with 'learning'".

Does it just return (and display) a link or does it actually take you to the queries page? (The latter makes more sense to me)

**Reply by srid** [reply]
Note: this can also be subsumed under #106 

> Does it just return (and display) a link or does it actually take you to the queries page?

Queries always render inline (they don't create a new dedicated page and then take you to it). In regards to what to display, it is best to let the user customize the HTML template for it, with Emanote providing a default one, as [described here](https://github.com/srid/emanote/discussions/141#discussioncomment-1370506).

**Reply by j-steinbach** [reply]
Then I would split my suggestion into two:

- Improvement of the queries
- An actual button (maybe some [dice](https://fontawesome.com/v5.15/icons?d=gallery&p=2&q=dice)) in the UI that takes you to a random, unspecified note
